pkg/controller/wip/controller.go

```diff
@@ -45,7 +45,7 @@ func NewController(
 	instanceInformer informers.InstanceInformer,
 	bindingInformer informers.BindingInformer,
 	brokerClientCreateFunc brokerapi.CreateFunc,
-) (Controller, error) {
+) (*controller, error) {
```

Comment:
This particular change isn't a good one to make. This makes the `NewController()` function return, very specifically a concrete type instead of an interface. This is bad because it limits the ability of those using this to code to the interface. It's also generally bad form for exported functions (like `NewController()`) to return non-exported types, because that will lead to dead ends in the godocs.

The right way around this is _not_ to make this change here, but rather to apply a type assertion in the test code so that you can coerce the `Controller` to a `*controller` to get access to all the impl-specific functions. (I'm certain that's why you did this.)